server/src/main/java/com/epam/aidial/core/server/util/RegexUtil.java

```diff
@@ -3,16 +3,23 @@
 import lombok.experimental.UtilityClass;
 
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Comparator;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 @UtilityClass
 public class RegexUtil {
 
-    public String replaceNamedGroups(Pattern pattern, String input, List<String> groups) {
-        if (groups == null || groups.isEmpty()) {
+    public String replaceNamedGroups(Pattern pattern, String input) {
+        return replaceNamedGroups(pattern, input, getNamedGroups(pattern));
+    }
+
+    public String replaceNamedGroups(Pattern pattern, String input, Collection<String> groups) {
+        if (pattern == null || input == null || input.isBlank() || groups == null || groups.isEmpty()) {
```

Comment:
We should check the first
`if (pattern == null || input == null || input.isBlank()` before making a list of groups